Set ForceNew for user_data fixing #42

jonasfj opened the pull request:
After reading the docs a bit I realize that if your provider doesn't support updating
a property the best solution is probably to label said property "ForceNew".

For user_data this is probably a rather desired property, since user_data is often used to
initialize the system. So force recreation of the resource might even be better than updating
user_data, which would still leave users wanting to have their machine reboot or reformatted
(in many instances).

t0mk commented:
Hi @jonasfj, thanks for the PR, it seems that user)data should be ForceNew indeed.

Did you actually try it, and confirmed that it fixes the use case which you described in the related issue?

It's pretty simple to build and install this provider, just see the readme in the root of this repo, and maybe the Makefile. I will be able to test this only on Monday myself.

jonasfj commented:
Okay, fair point... I was lazy and didn't test it... no excuse :)

FYI; the readme says to install it `GOPATH/bin`. But I don't see how that provides a version number, and nothing in terraform docs suggest that it picks up plugins from `PATH`, rather it says to install plugins in `.terraform.d/plugins` and naming them `terraform-provider-<name>_vX.Z.Y` (which worked). 

---
```
Terraform will perform the following actions:

-/+ packet_device.workers (new resource required)
      id:                      "YYYY" => <computed> (forces new resource)
      access_private_ipv4:     "10.80.180.131" => <computed>
      access_public_ipv4:      "147.75.83.141" => <computed>
      access_public_ipv6:      "2604:1380:2001:900::3" => <computed>
      always_pxe:              "false" => "false"
      billing_cycle:           "hourly" => "hourly"
      created:                 "2018-01-20T01:57:21Z" => <computed>
      facility:                "ams1" => "ams1"
      hardware_reservation_id: "" => <computed>
      hostname:                "tf-test" => "tf-test"
      locked:                  "false" => <computed>
      network.#:               "3" => <computed>
      operating_system:        "coreos_stable" => "coreos_stable"
      plan:                    "baremetal_0" => "baremetal_0"
      project_id:              "XXXX" => "XXXX"
      public_ipv4_subnet_size: "31" => <computed>
      root_password:           <sensitive> => <computed> (attribute changed)
      state:                   "active" => <computed>
      updated:                 "2018-01-20T02:02:03Z" => <computed>
      user_data:               <sensitive> => <sensitive> (forces new resource)


Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

packet_device.workers: Destroying... (ID: YYYY)
packet_device.workers: Destruction complete after 1s
packet_device.workers: Creating...
  access_private_ipv4:     "" => "<computed>"
  access_public_ipv4:      "" => "<computed>"
  access_public_ipv6:      "" => "<computed>"
  always_pxe:              "" => "false"
  billing_cycle:           "" => "hourly"
  created:                 "" => "<computed>"
  facility:                "" => "ams1"
  hardware_reservation_id: "" => "<computed>"
  hostname:                "" => "tf-test"
  locked:                  "" => "<computed>"
  network.#:               "" => "<computed>"
  operating_system:        "" => "coreos_stable"
  plan:                    "" => "baremetal_0"
  project_id:              "" => "XXXX"
  public_ipv4_subnet_size: "" => "<computed>"
  root_password:           "<sensitive>" => "<sensitive>"
  state:                   "" => "<computed>"
  updated:                 "" => "<computed>"
  user_data:               "<sensitive>" => "<sensitive>"
packet_device.workers: Still creating... (10s elapsed)
packet_device.workers: Still creating... (20s elapsed)
packet_device.workers: Still creating... (30s elapsed)
...
packet_device.workers: Still creating... (4m40s elapsed)
packet_device.workers: Creation complete after 4m49s (ID: 1e154648-21fd-454b-8826-7ef5d16813d8)

Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
```

It works :)

t0mk commented:
Thanks for testing @jonasfj. Looking at this, and then on Digital Ocean droplet resource at
https://github.com/terraform-providers/terraform-provider-digitalocean/blob/master/digitalocean/resource_digitalocean_droplet.go#L144
.. I see no reason not to merge this now.

Please note that to have this in the provider installed on `terraform init`, I still need to ask HashiCorp people to make a release. I was thinking of addig the description field and pushing #35 through for the next release, but if I won't manage this until next Friday 26.1, I will ask for release then anyway.

On Monday I will update the build and install instructions in the readme.
